src/components/App/custom-els/FileDrop/index.ts

```diff
@@ -110,15 +111,28 @@ export class FileDrop extends HTMLElement {
   private _onDrop(event: DragEvent) {
     event.preventDefault();
     this._reset();
-    const dragDataItem = firstMatchingItem(event.dataTransfer.items, this.accept);
-    if (!dragDataItem) return;
 
-    const file = dragDataItem.getAsFile();
+    const file = this._getFileData(event.dataTransfer);
     if (file === null) return;
 
     this.dispatchEvent(new FileDropEvent('filedrop', { file }));
   }
 
+  @bind
+  private _onPaste(event: ClipboardEvent) {
+    const file = this._getFileData(event.clipboardData);
+    if (file === null) return;
+
+    this.dispatchEvent(new FileDropEvent('filepaste', { file }));
+  }
+
+  private _getFileData(data: DataTransfer): File | null {
```

Comment:
wrt to interactions: if the UI element can't handle 'onpaste' it should be bubbling up to the file-drop element, if the element does handle `onpaste` then I would expect it to stop propagation and look to see if it can read the data.  I'll do some research to see what is happening with those elements.

wrt to focusable: that's actually a good point, we should probably make the other elements have the ability to focus, tapping outside of this edit box on github to the main page will remove focus of this element, and I think users would expect the same.

wrt to events: my thinking was that if you don't want to handle the event, you can just ignore the `onfilepaste` vs the event will fire and you have to check the event type... that being said, if you look at the internal logic, it's exactly the same so it might imply externally it would be seen the same.